process/executionTrack/executionResultsTracker.go

```diff
@@ -0,0 +1,165 @@
+package executionTrack
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+)
+
+type executionResultsTracker struct {
+	lastNotarizedResult    *block.ExecutionResult
+	mutex                  sync.RWMutex
+	executionResultsByHash map[string]*block.ExecutionResult
+	nonceHashes            *nonceHashes
+}
+
+// NewExecutionResultsTracker will create a new instance of *executionResultsTracker
+func NewExecutionResultsTracker() (*executionResultsTracker, error) {
+	return &executionResultsTracker{
+		executionResultsByHash: make(map[string]*block.ExecutionResult),
+		nonceHashes:            newNonceHashes(),
+	}, nil
+}
+
+// AddExecutionResult will add the provided execution result in tracker
+// It will return true if the execution result was added in the tracker
+func (est *executionResultsTracker) AddExecutionResult(executionResult *block.ExecutionResult) error {
+	if executionResult == nil {
+		return ErrNilExecutionResult
+	}
+
+	est.mutex.Lock()
+	defer est.mutex.Unlock()
+	if est.lastNotarizedResult == nil {
+		return ErrNilLastNotarizedExecutionResult
+	}
+
+	if est.lastNotarizedResult.Nonce >= executionResult.Nonce {
+		return fmt.Errorf("execution results nonce(%d) is lower than last notarized nonce(%d)", executionResult.Nonce, est.lastNotarizedResult.Nonce)
+	}
+
+	est.executionResultsByHash[string(executionResult.HeaderHash)] = executionResult
+	est.nonceHashes.addNonceHash(executionResult.Nonce, string(executionResult.HeaderHash))
+
+	return nil
+}
+
+// GetPendingExecutionResults will return the pending execution results
+func (est *executionResultsTracker) GetPendingExecutionResults() ([]*block.ExecutionResult, error) {
+	executionResults := make([]*block.ExecutionResult, 0)
```

Comment:
```suggestion
	executionResults := make([]*block.ExecutionResult, 0, len(est.executionResultsByHash))
```
and move it under the mutex protection

Reply:
fixed